signer/spiffe/spiffe.go

```diff
@@ -45,3 +54,96 @@ func Signer(ctx context.Context, socketPath string) (cryptoutil.Signer, error) {
 
 	return cryptoutil.NewSigner(svid.PrivateKey, cryptoutil.SignWithIntermediates(svid.Certificates[1:]), cryptoutil.SignWithCertificate(svid.Certificates[0]))
 }
+
+func DelgatedSigner(ctx context.Context, socketPath string, workloadPath string) (cryptoutil.Signer, error) {
```

Comment:
we probably should be checking array lengths before accessing them in this function to prevent panics